File: Java/SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {
        int[] arr ={38,52,9,18,6,62,13};
        selectionSort(arr);
        for (int i:arr) {
            System.out.print(i+" ");
        }
    }

    static void selectionSort(int[] arr){
        int min, temp;
        for (int i = 0; i < arr.length; i++) {
            min=i;
            for (int j = i+1; j < arr.length; j++) {
                if(arr[j]<arr[min]) min=j;
            }
            temp=arr[i];
            arr[i]=arr[min];
            arr[min]=temp;
        }
    }
}

File: C++/Random_programs/Selection_Sort.cpp
#include <bits/stdc++.h>
using namespace std;

void selectionSort(int arr[], int n)
{
    int i=0, j=0, min_idx;
    for (i = 0; i < n-1; i++)
    {
        min_idx = i;
        for (j = i+1; j < n; j++)
        {
            if (arr[j] < arr[min_idx])
            {
                min_idx = j;
            }
        }
        swap(arr[min_idx], arr[i]);
    }
}
int main()
{
    int n;
    cin>>n;
    int a[n];
    for(int i=0;i<n;i++)cin>>a[i];

    selectionSort(a, n);

    for(int i=0;i<n;i++)cout<<a[i]<<" ";
}
